### DISCUSSION TITLE
Inverted Display

### DISCUSSION THREAD

**Discussion by VogonCaptain** [harbaum/galagino#45] [Ideas]
For some reason, my display had all the colors inverted and I couldn't figure out how to fix it within the software as written.  I'm not very familiar with how GitHub works to suggest code changes but I was able to work a fix by adding the following lines:

Config.h:
43 #define TFT_INVERT    //uncomment to invert display

video.cpp:
88   #ifdef TFT_INVERT
89   0x21,                             //invert display
90   #endif

in case any of the readers of this post can't find where to put it because the line numbers are off, these are added right before the line that says:
0x00                              // End of list

in the file video.cpp.  In config.h the line can go anywhere.

Hopefully this helps someone!

**Comment by jimxbob** [comment]
I had a similar problem with my screen - the colours look like they're all inverted.
It's an ST7789.
Looking at the line numbers of the code above, I think that that was inserted in the code for the ILI9341.
I tried inserting it at the end of the ST7789 init_cmd[] in vidoe.cpp but it didn't help. Looking more closely at this, at line 104 we have:
  0x21, 0,                          // INV ON
I tried changing this to (guesswork)
  0x21, 1,                        
But that didn't work - it just gave me a blank white screen.
HOWEVER - commenting out that line 104 did the trick!
 // 0x21, 0,                          // INV ON
I'm leaving this here as the first comment set me going in the right direction, and I hope this help another get there quicker!!

**Reply by reyco2000** [reply]
Two magic lines I guess will be good to set a var in config.h to invert ST7789,
  0x20, 0,                          // INV OFF
  
  Original line
  0x21, 0, // INV ON
